packages/core/src/browser/shell/sidebar-bottom-menu-widget.tsx

```diff
@@ -0,0 +1,77 @@
+/********************************************************************************
+ * Copyright (C) 2020 Alibaba Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import * as React from 'react';
+import { ReactWidget } from '../widgets';
+import { ContextMenuRenderer } from '../context-menu-renderer';
+import { MenuPath } from '../../common/menu';
+
+export const SidebarBottomMenuWidgetFactory = Symbol('SidebarBottomMenuWidgetFactory');
+
+export interface SidebarBottomMenu {
+  id: string;
+  iconClass: string;
+  title: string;
+  menuPath: MenuPath;
+  order: number; // smaller one place lower
+}
+
+/**
+ * The menu widget placed on the bottom of the sidebar.
+ */
+@injectable()
+export class SidebarBottomMenuWidget extends ReactWidget {
+  protected readonly menus: SidebarBottomMenu[];
+
+  @inject(ContextMenuRenderer)
+  protected readonly contextMenuRenderer: ContextMenuRenderer;
+
+  constructor() {
+    super();
+    this.menus = [];
+  }
+
+  addMenu(menu: SidebarBottomMenu): void {
```

Comment:
returning Disposable would be good to unregister and having a method like removeMenu by id, but it could be done later when someone requested

Reply:
maybe later :p